src/Actions/MockActionResponse.php

```diff
@@ -108,35 +111,51 @@ public function assertVisit(string $path = '', array $options = []): self
             return $this->assertResponseType('visit');
         }
 
-        return $this->assertResponseContainsArray(['path' => $path, 'options' => array_values($options)], 'visit');
+        PHPUnit::assertArrayHasKey('visit', $this->response);
+
+        $visit = $this->response['visit'];
+
+        PHPUnit::logicalAnd(
+            PHPUnit::assertNotEmpty($visit),
+            PHPUnit::assertInstanceOf(Visit::class, $visit),
+            PHPUnit::assertEquals($options, $visit->options),
+            PHPUnit::assertEquals($path, $visit->path),
+        );
+
+        return $this;
     }
 
     /**
      * Asserts the handle response is of type "openInNewTab".
      *
-     * @param  string  $message
+     * @param null|string $path
+     * @param string $message
      * @return $this
      */
-    public function assertOpenInNewTab(string $message = ''): self
+    public function assertOpenInNewTab(?string $path = null, string $message = ''): self
     {
-        return $this->assertResponseType('openInNewTab', $message);
+        if (blank($path)) {
+            return $this->assertResponseContainsArray(['openInNewTab' => true], 'redirect', $message);
+        }
+
+        return $this->assertResponseContainsArray(['url' => $path, 'openInNewTab' => true], 'redirect', $message);
     }
 
     /**
      * Asserts the handle response is of type "download".
      *
-     * @param  string  $message
+     * @param string $message
      * @return $this
      */
     public function assertDownload(string $message = ''): self
     {
         return $this->assertResponseType('download', $message);
     }
 
-    private function assertResponseContains(string $contents, string $type, string $message = ''): self
+    private function assertResponseKeyContains(string $contents, string $type, string $key, string $message = ''): self
```

Comment:
Why did this need to be changed?

Reply:
`assertResponseContains` was orphaned.

`assertResponseKeyContains` was required as the underlying composition of actions changed in Nova 5.

Specifically: In nova 4 the action response for `Action::message()` was just the message string, in Nova 5 it is a class that has a member called `text` where the message string is contained.

This same change was mimicked in `Action::danger()`